modules/TransitionUtils.js

```diff
@@ -41,6 +42,10 @@ export function runEnterHooks(routes, nextState, callback) {
 
   let redirectInfo
   function replaceState(state, pathname, query) {
+    warning(
```

Comment:
Drop this - we'll handle it separately. This entire function should just take a location descriptor.